### ISSUE TITLE
Remotebuild requiring cordova-ios 4.3.0

### ISSUE THREAD

**Issue by XactaAndy** [Microsoft/remotebuild#44]
I'm having trouble getting my Mac to build my cordova project since XCode updated to version 8.3. I'm using version 6.1.1 of the cordova CLI. Versions higher than this have historically given me different errors.

Here is a copy of the [build.log.txt](https://github.com/Microsoft/remotebuild/files/1094427/build.log.txt). This is the error that is reported back in VS2015 "Error: Remote build error from the build server Build failed with error Remotebuild requires your projects to use cordova-ios 4.3.0 or greater with XCode 8.3. Please update your cordova-ios version. - 1"

I have followed the advice from issue #26 and cleared out the ~/.taco_home/node_modules folder but the issue still persists. 

Any ideas on how to get round this issue?

**Comment by Webrusterkk** [comment]
I got the same issue in the past , I have installed Cordova 6.5.0 to make it work

**Comment by XactaAndy** [comment]
If I try and build CLI v6.5.0 I get the following error "Error: Remote build error from the build server Error: Failed to acquire Cordova version 6.5.0. Error: EACCES, permission denied '/Users/Xactaware/.config/configstore/insight-cordova.json'"

**Comment by Webrusterkk** [comment]
It's due to permission issue ..  you need to give the elevated permission .. same was discussed in the TACO site. Here is the link which saves your time

http://taco.visualstudio.com/en-us/docs/tips-and-workarounds-ios-readme/#permission-issues-eacces-errors

**Comment by dmitrii-snytkin** [comment]
@XactaAndy Thank you for your feedback. Has advise of @Webrusterkk solved your problem?

**Comment by XactaAndy** [comment]
@dmitrii-snytkin Unfortunately it hasn't fixed my issue. I don't have the platforms folder checked into TFS, however my plugins folder is. I did try the fix of adding in the hooks workaround but the issue still persists. If I try and navigate to .config folder on the mac it doesn't exist. Is this folder created by the remotebuild or another program?

**Comment by Webrusterkk** [comment]
@XactaAndy  did you tried to install the Cordova in Mac with `sudo` ? and version need to be `cordova 6.5.0`

**Comment by XactaAndy** [comment]
@Webrusterkk Yes, I just ran "sudo npm install cordova@6.5.0" again. I still get the same error when I try and build via VS 2015.

**Comment by XactaAndy** [comment]
I have managed to fix the error. There must have been some odd permissions on some of the folders as running this fixed it for me:

```
sudo chown -R $USER:$GROUP ~/.npm
sudo chown -R $USER:$GROUP ~/.config
```

**Comment by Webrusterkk** [comment]
@XactaAndy  yes, this another way to give the elevated permission in MAC as provided in this issue #16 .
Hope you made it .. @dmitrii-snytkin  you can close this issue. and it would be better to add this also to the docs ..

**Comment by dmitrii-snytkin** [comment]
@Webrusterkk @XactaAndy  Thank you for your help! I will suggest to add this case in the documentation.

**Comment by johnploper** [comment]
I am also getting the build error 'Build failed with error Remotebuild requires your projects to use cordova-ios 4.3.0 or greater with XCode 8.3.'.  I tried reinstalling with sudo ('sudo npm install cordova@6.5.0').  I also tried 'sudo npm install cordova@7.0.0'.  I also tried the chown command listed above.  I reinstalled VS 2017, and upgraded to version 15.3.  None of this works.  I see in my config.xml that cordova-ios version 4.2 is being used.  I don't see how to change/select version 4.3, which seems to be what is required.

Any idea what to try next ?
